drivers/can/can_native_linux_adapt.c

```diff
@@ -113,7 +113,7 @@ int linux_socketcan_poll_data(int fd)
 	return -EAGAIN;
 }
 
-ssize_t linux_socketcan_read_data(int fd, void *buf, size_t buf_len, bool *msg_confirm)
```

Comment:
This change seems unrelated to the general topic, and also seems wrong -- the underlying issue is the incorrect cast of the ssize_t return from recvmsg to int.

Reply:
They are related in the sense that ssize_t is an extension. So either ssize_t is not used, or this files set POSIX_C_SOURCE too (note the proto is a header used by "the embedded side").
It is correct that recvmsg (or write or read) can in theory return something in the whole range of ssize_t, but given the use, the maximum they will return is the size of the buffer sent or received which cannot be bigger than the packet size for the protocols which I do not know from the top of my head, but I'm guessing not bigger than IP's 64KB, so the int range should be more than sufficient.
That's why I just took the easier approach of simple having these return int (as the users anyhow also just cast to int), instead of adding POSIX_C_SOURCE to these embedded side of these drivers.

Does that sound sensible enough?

Reply:
Another thing I forgot yesterday, these functions are built with one C library and called from another.
ssize_t being a C library defined type can in principle not match between both, so that would also speak for only using basic types defined by the compiler.
CC: @jukkar @henrikbrixandersen as authors of this code. Your input is also welcome :)

Reply:
I am fine with changing the function signature as this indeed could cause issues. To make the intention clear, perhaps add an explicit cast to int for the return value of the`recvmsg()` call?